### PR TITLE
fix: cascader component value support number type

### PR DESCRIPTION
**Pull request by Loogeek**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/17407

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix Cascader not support `number[]` value.       |
| 🇨🇳 Chinese |     修复 Cascader 不支持 `number[]` 类型 `value` 的问题。      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/cascader/index.en-US.md](https://github.com/Loogeek/ant-design/blob/cascader-value-type/components/cascader/index.en-US.md)
[View rendered components/cascader/index.zh-CN.md](https://github.com/Loogeek/ant-design/blob/cascader-value-type/components/cascader/index.zh-CN.md)

### CODE REVIEW

**Comment by afc163** [comment]
类似 Form 来个 `type.test.tsx` 做 ts 定义的测试用例吧。

**Comment by Loogeek** [comment]
> 类似 Form 来个 `type.test.tsx` 做 ts 定义的测试用例吧。
不是太明白项目中几个type.test.tsx测试用例的作用，加了下，看看是否可以

**Comment by afc163** [comment]
> 不是太明白项目中几个type.test.tsx测试用例的作用，加了下，看看是否可以

就是测试 tsc 编译是否能过，你故意写错然后跑一下 `npm test -- components/cascader` 就能测试出来了。

**Comment by Loogeek** [comment]
> 就是测试 tsc 编译是否能过，你故意写错然后跑一下 `npm test -- components/cascader` 就能测试出来了。

是的，我有试过，比如result的type.test加了个数值的prefixCls，看test都还是过的

**Comment by afc163** [comment]
`npm run tsc` 应该能检查出来。

**Comment by Loogeek** [comment]
> npm run tsc

恩恩，tsc能监测出来

File: components/cascader/index.en-US.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ Cascade selection box.
 | size | input size | `large` \| `middle` \| `small` |  |  |
 | style | additional style | CSSProperties | - |  |
 | suffixIcon | The custom suffix icon | ReactNode | - |  |
-| value | selected value | string\[] | - |  |
+| value | selected value | string\[] | number\[] | - |  |
```
Comment (yoyo837) [inline components/cascader/index.en-US.md:45]:
```diff
- | defaultValue | initial selected value | string\[] | number\[] | \[] |  |
+ | defaultValue | initial selected value | string\[] \| number\[] | \[] |  |
```

File: components/cascader/index.en-US.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ Cascade selection box.
 | bordered | whether has border style | boolean | true |  |
 | changeOnSelect | change value on each selection if set to true, see above demo for details | boolean | false |  |
 | className | additional css class | string | - |  |
-| defaultValue | initial selected value | string\[] | \[] |  |
+| defaultValue | initial selected value | string\[] | number\[] | \[] |  |
```
Comment (yoyo837) [inline components/cascader/index.en-US.md:28]:
```diff
- | defaultValue | initial selected value | string\[] | number\[] | \[] |  |
+ | defaultValue | initial selected value | string\[] \| number\[] | \[] |  |
```

**Pull request merged**